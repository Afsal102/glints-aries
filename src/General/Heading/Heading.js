/* @flow */

import React, { Fragment } from 'react';
import { HeadingContainer } from '../../Style/HeadingStyle';

const Heading = (props: Props) => {
  const {
    className,
    inline,
    children,
    ...defaultProps
  } = props;

  return (
    <Fragment>
      <HeadingContainer className={className} inline={inline} {...defaultProps}>
        <span>
          {children}
        </span>
      </HeadingContainer>
    </Fragment>
  );
};

type Props = {
  children: React$Node,
  className: string,
  inline: boolean,
}

export default Heading;
