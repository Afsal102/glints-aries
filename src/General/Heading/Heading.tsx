import * as React from 'react';
import { HeadingContainer } from '../../Style/General/HeadingStyle';

const Heading: React.FunctionComponent<Props> = (props) => {
  const {
    className,
    inline,
    children,
    uppercaseText,
    ...defaultProps
  } = props;

  return (
    <React.Fragment>
      <HeadingContainer
        id="aries-heading"
        className={className}
        inline={inline}
        uppercaseText={uppercaseText}
        {...defaultProps}
      >
        <span id="heading-text">
          {children}
        </span>
      </HeadingContainer>
    </React.Fragment>
  );
};

interface Props extends React.ComponentPropsWithoutRef<typeof HeadingContainer> {
  children: React.ReactNode;
}

export default Heading;
