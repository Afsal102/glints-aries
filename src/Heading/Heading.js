/* @flow */

import React, { Fragment } from 'react';
import { HeadingLarge, HeadingMedium, HeadingSmall } from '../Style/HeadingStyle';
import { Size } from '../Utils/StyleConfig';

const renderHeading = ({
  children,
  size,
  className,
  ...defaultProps
}: Props) => {
  switch (size) {
    case Size.LARGE:
      return (
        <HeadingLarge className={className} {...defaultProps}>
          {children}
        </HeadingLarge>
      );
    case Size.MEDIUM:
      return (
        <HeadingMedium className={className} {...defaultProps}>
          {children}
        </HeadingMedium>
      );
    case Size.SMALL:
      return (
        <HeadingSmall className={className} {...defaultProps}>
          {children}
        </HeadingSmall>
      );
    default:
      return (
        null
      );
  }
};

const Heading = (props: Props) => (
  <Fragment>
    {renderHeading(props)}
  </Fragment>
);

type Props = {
  children: React$Node,
  size: string,
  className: string,
}

export default Heading;
