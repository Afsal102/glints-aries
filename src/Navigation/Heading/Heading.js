/* @flow */

import React, { Fragment } from 'react';
import { Container } from '../../Style/HeadingStyle';

const Heading = (props: Props) => {
  const {
    className,
    inline,
    children,
    ...defaultProps
  } = props;

  return (
    <Fragment>
      <Container className={className} inline={inline} {...defaultProps}>
        <span>
          {children}
        </span>
      </Container>
    </Fragment>
  );
};

type Props = {
  children: React$Node,
  className: string,
  inline: boolean,
}

export default Heading;
