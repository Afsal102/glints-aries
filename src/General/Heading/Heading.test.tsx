import * as React from 'react';

import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Heading from './Heading';

const heading = 'This is a Heading';

it(`<Heading> should match snapshot when the component is called`, () => {
  const HeadingSnapshot = renderer
    .create(<Heading className="test-class-name">{heading}</Heading>)
    .toJSON();
  expect(HeadingSnapshot).toMatchSnapshot();
});

describe('<Heading> should render the correct case for the text when', () => {
  it('uppercaseText is true', () => {
    const { getByRole } = render(<Heading uppercaseText>{heading}</Heading>);
    const headingContainer = getByRole('heading');
    expect(headingContainer).toHaveStyle('text-transform: uppercase');
  });

  it('uppercaseText is false', () => {
    const { getByRole } = render(
      <Heading uppercaseText={false}>{heading}</Heading>
    );
    const headingContainer = getByRole('heading');
    const { textTransform } = getComputedStyle(headingContainer);
    expect(textTransform).toBeFalsy();
  });
});

describe('<Heading> should render the correct display for the text when', () => {
  it('inline is true', () => {
    const { getByRole } = render(<Heading inline>{heading}</Heading>);
    const headingContainer = getByRole('heading');
    expect(headingContainer).toHaveStyle('display: inline-block');
  });

  it('inline is false', () => {
    const { getByRole } = render(<Heading inline={false}>{heading}</Heading>);
    const headingContainer = getByRole('heading');
    expect(headingContainer).toHaveStyle('display: block');
  });
});
