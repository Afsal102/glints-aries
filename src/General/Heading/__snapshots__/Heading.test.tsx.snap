// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> should match snapshot when the component is called 1`] = `
<h1
  className="HeadingStyle__HeadingContainer-sc-1u1nmvn-0 fnrjID aries-heading"
  role="heading"
>
  <span
    className="heading-text"
  >
    This is a Heading
  </span>
</h1>
`;
